### PR TITLE
feature/authentication-module/add reset password component

### PR DESCRIPTION
**Pull request by jahammo2**
#### Why
Because these are common copy-and-paste features on front end projects.

#### What
- Added resetPassword component and container
- Had to add a rule to the linter because `['access-token']` was throwing a linter. I think it's best I don't just use the `eslint-ignore` because this is pretty common. I'll put in a PR to the SB linter too

### CODE REVIEW

File: modules/authentication/components/ResetPassword.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const values = mapFormValues(this.resetPasswordForm.elements);
+
+    const resetPasswordOptions = {
+      ['access-token']: this.props.location.query.token,
```
Comment (zachary-kuhn) [inline modules/authentication/components/ResetPassword.jsx]: Why do `['access-token']` and not `'access-token'`?

Comment (jahammo2) [inline modules/authentication/components/ResetPassword.jsx]: Lol I didn't know you could do that

File: modules/authentication/components/ResetPassword.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const values = mapFormValues(this.resetPasswordForm.elements);
+
+    const resetPasswordOptions = {
```
Comment (greg5green) [inline modules/authentication/components/ResetPassword.jsx]: If we're allowing additional children to be passed into this component, this really should be
```
const resetPasswordOptions = {
  ...values,
  'access-token': this.props.location.query.token,
  client: this.props.location.query.client_id,
  uid: this.props.location.query.uid
};
```

Comment (jahammo2) [inline modules/authentication/components/ResetPassword.jsx]: I like it

File: .eslintrc [hunk 3]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": "smashing-boxes/react"
+  "extends": "smashing-boxes/react",
+  "rules": {
+    "no-useless-computed-key": 0           // http://eslint.org/docs/rules/no-useless-computed-key
```
Comment (jahammo2) [inline .eslintrc]: 👍

**Pull request merged**